lib/facility.rb

```diff
@@ -1,14 +1,146 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_info)
+    @name                = facility_info[:name]
+    @address             = facility_info[:address]
+    @phone               = facility_info[:phone]
+    @services            = []
+    @registered_vehicles = []
+    @collected_fees      = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if self.services.include?("Vehicle Registration")
+    @registered_vehicles << vehicle
```

Comment:
Make sure to indent the code that's inside the if block.

Alternatively, although this looks great and is very readable, if you wanted to get rid of that if/end block, you could use what we call a guard clause. This is a one line if statement to check a condition before we even run the rest of the method. In this case, it would look like `return nil if !services.include?("Vehicle Registration")`. then you wouldn't need the if/end block.

Also, we don't need the `self` keyword when calling a method that exists within the class we're currently in. It's implied.